content/blog/2020-07-31-spreadsheet-workflows-using-r/index.Rmarkdown

```diff
@@ -0,0 +1,185 @@
+
+
+## Nested Logic
+One advantage of spreadsheets is we can use formulas to dictate what should be happening in a specific cell and in a broader application of this, apply them to an entire column.  When creating a column that requires multiple or 'nested' logical conditions, we can take advantage of the `case_when()` function from dplyr.  Similar to a CASE statment in SQL we can set parameters for what a value should be based off conditions from other rows or columns in our data.
+
+Back to our spending data, while data for all 12 months is useful for our summary we want to take a look at spending by quarter, something not provided in the initial dataset.  
+To do this we can create a new column in our dataset using the `mutate()` function and apply `case_when()` within that call.  
+```{r}
+spending_quarter <- spending %>%
+  mutate(quarter = case_when(month %in% c("Jan", "Feb", "Mar") ~ "Q1",
+                             month %in% c("Apr", "May", "Jun") ~ "Q2",
+                             month %in% c("Jul", "Aug", "Sep") ~ "Q3",
+                             month %in% c("Oct", "Nov", "Dec") ~ "Q4",
+                             TRUE ~ "NA"))
+
+spending_quarter
+```
+
+Excellent! We've got our new quarter column set up. Here we specified which months make up each quarter and for good mweasure added an NA condition in the event we've missed something. We can quality check our case_when by examining our data and seeing if any NA values show up which would indicate if a month is spelled differently or if there's an error in our code.  
+
+
+## Pivot to this section
+
+Pivot tables are a powerful tool spreadsheets offer, allowing us to generate summaries of large quantities of data. Using some data wrangling functions from the tidyverse we can create summary tables as well. Let's take a look total expenses and budget by expense type and quarter.  
+```{r}
+summary_table <- spending_quarter %>%
+  group_by(expense, quarter) %>%
+  summarise(total_expenses = sum(num_of_expenses),
+            total_expense_amt = sum(amount),
+            total_budget_amt = sum(budget),
+           .groups = "drop")
+
+summary_table
+```
+
+## Row Calculations
+
+One thing that can be tricky when working with pivot tables is calculating new variables by rows as opposed to columns, with a usual work around being to manually add in a formula in a cell adjacent to our table.  Working with our data in R we can take advantage of the `rowwise()` function from dplyr, telling R we want to compute new values and summary statistics by row and not column.  
+Let's take a look at how much money was left over in each quarter by subtracting money spent from our budget and use the `rowwise()` function to specify we want this done by each row in our summary table
+```{r}
+left_over <- summary_table %>%
+  rowwise() %>%
```

Comment:
I'm not running this code. Or rather, I'm only running it in my head. But it's not obvious to me that using `rowwise()` has an effect here.

Reply:
Would it make sense to explain the function a bit more? I'm a bit keen on keeping this in as I just think its so useful and really want to showcase this in the example. As I was doing the spreadsheet comparison to add here I was even reminded of how useful it was with it not available to me.

Reply:
Happy to remove this however I think we'd also have to remove the visualization section as its directly tied to the row calculation section.